Controller/Proxy/ProxyController.cs

```diff
@@ -0,0 +1,36 @@
+﻿using Castle.DynamicProxy;
+using Model.Database.Contexts;
+
+namespace Controller.Proxy
+{
+    public static class ProxyController
+    {
+        private static ProxyGenerator generator = new ProxyGenerator();
+
+        /**
+         * Add a class to the generator proxy
+         *
+         * @param parameters Parameters to use for the class' constructor
+         * @param context The IDatabaseContext to use for the TracingInterceptorAspect
+         *
+         * @return instance of class T
+         */
+        public static T AddToProxy<T>(object[] parameters, IDatabaseContext context) where T : class
+        {
+            return (T) generator.CreateClassProxy(typeof(T), parameters, new TracingInterceptorAspect(context));
+        }
+
+        /**
+         * Add a class to the generator proxy
+         *
+         * @param context The IDatabaseContext to use for the TracingInterceptorAspect
+         *
+         * @return instance of class T
+         */
+        public static T AddToProxy<T>(IDatabaseContext context) where T : class
+        {
+            return generator.CreateClassProxy<T>(new TracingInterceptorAspect(context));
+        }
+

```

Comment:
Wat is het verschil eigenlijk in de twee functies, aangezien je in de bovenste twee keer context meegeeft

Reply:
De een neem arguments mee om aan de constructor mee te geven.
De ander alleen maar de context.
En die context heeft die TracingInterceptorAscpect nodig, om uit de huidige database context permissions op te halen.

Vandaan, en toevallig hebben al die controllers ook de database context als argument nodig.